### PR TITLE
fix(webhooks): [BUG] Change webhook type for invoice retry action

### PR DESCRIPTION
**Pull request by lovrocolic**
## Description

When performing invoice retry action wrong webhook type was passed as parameter

### CODE REVIEW

File: app/services/invoices/payments/retry_service.rb [hunk 1]
```diff
@@ -32,9 +32,14 @@ def call
       delegate :customer, to: :invoice
 
       def webhook_type
-        return :invoice if invoice.invoice_type == 'subscription'
-
-        invoice.invoice_type.to_sym
+        case invoice.invoice_type
```
Comment (jdenquin) [inline app/services/invoices/payments/retry_service.rb:35]: Style/HashLikeCase: Consider replacing `case-when` with a hash lookup.

File: app/services/invoices/payments/retry_service.rb [hunk 2]
```diff
@@ -17,7 +23,7 @@ def call
           return result.not_allowed_failure!(code: 'payment_processor_is_currently_handling_payment')
         end
 
-        SendWebhookJob.perform_later(webhook_type, invoice) if customer&.organization&.webhook_url?
+        SendWebhookJob.perform_later(WEBHOOK_TYPE[invoice.invoice_type], invoice) if customer&.organization&.webhook_url?
```
Comment (jdenquin) [inline app/services/invoices/payments/retry_service.rb:26]: Style/IfUnlessModifier: Modifier form of `if` makes the line too long. (https://rubystyle.guide#if-as-a-modifier)

Comment (jdenquin) [inline app/services/invoices/payments/retry_service.rb:26]: Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)

File: app/services/invoices/payments/retry_service.rb [hunk 3]
```diff
@@ -17,7 +23,9 @@ def call
           return result.not_allowed_failure!(code: 'payment_processor_is_currently_handling_payment')
         end
 
-        SendWebhookJob.perform_later(webhook_type, invoice) if customer&.organization&.webhook_url?
+        if customer&.organization&.webhook_url?
```
Comment (vincent-pochet) [inline app/services/invoices/payments/retry_service.rb:26]: What do you think of extracting this block in a `deliver_webhook` method?

**Review by vincent-pochet — APPROVED** [review]

**Review by rsempe — APPROVED** [review]

File: app/services/invoices/payments/retry_service.rb [hunk 4]
```diff
@@ -31,10 +37,10 @@ def call
 
       delegate :customer, to: :invoice
 
-      def webhook_type
-        return :invoice if invoice.invoice_type == 'subscription'
+      private
```
Comment (jdenquin) [inline app/services/invoices/payments/retry_service.rb:40]: Lint/UselessAccessModifier: Useless `private` access modifier.

File: app/services/invoices/payments/retry_service.rb (hunk 3 above)
Comment (lovrocolic) [reply app/services/invoices/payments/retry_service.rb]: extracted

**Pull request merged**